spacy_llm/tasks/entity_linker/util.py

```diff
@@ -0,0 +1,190 @@
+            mentions=mentions,
+            entity_descriptions=[
+                [ent.description for ent in ents] for ents in cands_ents
+            ],
+            entity_ids=[[ent.id for ent in ents] for ents in cands_ents],
+            solutions=solutions,
+            reasons=[""] * len(mentions),
+        )
+
+
+def score(examples: Iterable[Example], **kwargs) -> Dict[str, Any]:
+    """Score entity linking accuracy in examples.
+    examples (Iterable[Example]): Examples to score.
+    RETURNS (Dict[str, Any]): Dict with metric name -> score.
+    """
+    return Scorer.score_links(examples, negative_labels=[EntityLinker.NIL])
+
+
+def ent_desc_reader_csv(path: Union[Path, str]) -> Dict[str, str]:
+    """Instantiates entity description reader with two columns: ID and description.
+    path (Union[Path, str]): File path.
+    RETURNS (Dict[str, str]): Dict with ID -> description.
+    """
+    with open(path) as csvfile:
+        descs: Dict[str, str] = {}
+        for row in csv.reader(csvfile, quoting=csv.QUOTE_ALL, delimiter=";"):
+            if len(row) != 2:
+                continue
+            descs[row[0]] = row[1]
+
+        if len(descs) == 0:
+            raise ValueError(
+                "Format of CSV file with entity descriptions is wrong. CSV has to be formatted as "
+                "semicolon-delimited CSV with two columns. The first columns has to contain the entity"
+                " ID, the second the entity description."
+            )
+
+    return descs
+
+
+@dataclasses.dataclass
+class BaseInMemoryLookupKBLoader:
+    path: Union[str, Path]
+    """Path to artefact containing knowledge base data."""
+
+    def __post_init__(self):
+        if isinstance(self.path, str):
+            self.path = Path(self.path)
+
+    @abc.abstractmethod
+    def __call__(self, vocab: Vocab) -> Tuple[InMemoryLookupKB, DescFormat]:
+        """Loads KB instance.
+        vocab (Vocab): Vocab instance of executing pipeline.
+        RETURNS (Tuple[InMemoryLookupKB, DescFormat]): Loaded/generated KB instance; descriptions for entities.
+        """
+        ...
+
+
+@dataclasses.dataclass
+class KBSerializedLoader(BaseInMemoryLookupKBLoader):
+    """Config/init helper class for loading InMemoryLookupKB instance from a serialized KB directory."""
+
+    nlp_path: Optional[Union[str, Path]]
+    """Path to serialized NLP pipeline. If None, path will be guessed."""
+    desc_path: Optional[Union[Path, str]]
+    """Path to .csv file with descriptions for entities. Has to have two columns with the first one being the entity ID,
+        the second one being the description. The entity ID has to match with the entity ID in the stored knowledge
+        base.
+        If not specified, all entity descriptions provided in prompts will be a generic "No description available" or
+        something else to this effect.
+    """
+    ent_desc_reader: EntDescReader
+    """Entity description file reader."""
```

Comment:
The formatting of this docstring is not as we usually do it

Reply:
Better in https://github.com/explosion/spacy-llm/pull/241/commits/f9991d57f5db7fe5159f641825f391cb348ca309?